### PR TITLE
Sveltekit middleware

### PR DESCRIPTION
**Pull request by GuiBibeau**
### Description

Adds a simple middleware example on svelte kit with description

### Demo URL
https://svelte-middleware-guivercel.vercel.sh/

### Type of Change

- [x] New Example
- [ ] Example updates (Bug fixes, new features, etc.)
- [ ] Other (changes to the codebase, but not to examples)

### New Example Checklist

- [ ] 🛫 `npm run new-example` was used to create the example
- [ ] 📚 The template wasn't used but I carefuly read the [Adding a new example](https://github.com/vercel/examples#adding-a-new-example) steps and implemented them in the example
- [ ] 📱 Is it responsive? Are mobile and tablets considered?

### CODE REVIEW

**Review by lfades — CHANGES_REQUESTED** [review]
- Running the example locally with `vc dev` shows a type error
- The use case for middleware here isn't showcased properly, it's about clicking a link to a redirect to the same page, providing no visual feedback of anything more than a loading spinner that makes it feel like it's broken, I think we can do better here by doing a redirect to a different page with a text saying something like "You were redirected by Middleware to /about/new from /about"
- The local build was throwing errors for me

I have updated some of the configuration files to make them more friendly.

File: edge-functions/svelte-middleware/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# SvelteKit With Middleware
+
+This example shows how use a Vercel Middleware in Sveltekit
```
Comment (Rich-Harris) [inline edge-functions/svelte-middleware/README.md:3]:
```suggestion
This example shows how use Vercel Middleware with SvelteKit
```

File: edge-functions/svelte-middleware/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+## Demo
+
+https://svelte-middleware-guivercel.vercel.sh/
+
+## How to Use
+
+- Install the Vercel CLI `npm i -g vercel`
+- Install degit `npm i -g degit`
+- Clone the repository `degit https://github.com/vercel/examples/edge-functions/svelte-middleware svelte-middleware`
```
Comment (Rich-Harris) [inline edge-functions/svelte-middleware/README.md:13]: this doesn't seem to actually work. degit only understands the repo URL itself, not subdirectories (it should probably error on encountering an argument like this)

```suggestion
- Install the Vercel CLI `npm i -g vercel`
- Clone the repository `npx degit https://github.com/vercel/examples vercel-examples`
- Navigate to the directory `cd vercel-examples/edge-functions/svelte-middleware`
```

File: edge-functions/svelte-middleware/src/routes/blocked.svelte [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<script context="module">
+  // disable client-side router to let requests be intercepted by Middleware
+  export const router = false
+</script>
+
+<h1 class="h1">Vercel Middleware + Sveltekit</h1>
```
Comment (Rich-Harris) [inline edge-functions/svelte-middleware/src/routes/blocked.svelte:6]:
```suggestion
<h1 class="h1">Vercel Middleware + SvelteKit</h1>
```

File: edge-functions/svelte-middleware/src/routes/blocked.svelte [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<script context="module">
+  // disable client-side router to let requests be intercepted by Middleware
+  export const router = false
+</script>
```
Comment (Rich-Harris) [inline edge-functions/svelte-middleware/src/routes/blocked.svelte:4]: Can we maybe come up with a different example that doesn't require this — something like rewriting text or swapping out an image? IMHO it's a bit confusing if someone has to disable SvelteKit features in order to see the middleware do something. Redirects would normally be handled within the framework

**Comment by Rich-Harris** [comment]
Should this work with `vc dev`? It doesn't for me, with 25.1.1-canary.11

**Pull request closed without merging**